File: 디자인패턴/java 디자인 패턴/Mediator/Door.java
public class Door extends Participant {
    private boolean bClosed = false;
    
    public Door(Mediator mediator) {
        super(mediator);
    }

    public void Open() {
        bClosed = true;
        mediator.participantChange(this);
    }

    public void Close() {
        bClosed = false;
        mediator.participantChange(this);
    }

    public boolean isDoor() {
        return bClosed;
    }

    @Override
    public String toString() {
        if(bClosed) return "# 문: 열림";
        else return "# 문: 닫힘";
    }
}

File: 디자인패턴/Python 디자인 패턴/Mediator/Door.py
from Participant import*

class Door(Participant):
    def __init__(this, mediator: Mediator) -> None:
        super().__init__(mediator)
        this.bDoor = False
    
    def Open(this):
        this.bDoor = True
        this.mediator.participantChange(this)
    
    def Close(this):
        this.bDoor = False
        this.mediator.participantChange(this)
    
    def isDoor(this):
        return this.bDoor
    
    def strPrint(this):
        if(this.bDoor):return "# 문 열림"
        return "# 문 닫힘"
    pass